src/Jaeger/Senders/HttpSender.cs

```diff
@@ -45,15 +46,16 @@ private HttpSender(Builder builder)
             }
 
             _transport = new THttpClientTransport(collectorUri, customHeaders);
-            _agentClient = new Agent.Client(ProtocolFactory.GetProtocol(_transport));
+            _protocol = ProtocolFactory.GetProtocol(_transport);
         }
 
         protected override async Task SendAsync(ThriftProcess process, List<ThriftSpan> spans, CancellationToken cancellationToken)
         {
             try
             {
                 var batch = new ThriftBatch(process, spans);
-                await _agentClient.emitBatchAsync(batch, cancellationToken).ConfigureAwait(false);
```

Comment:
The existing UdpSender uses the same logic and that works fine, doesn't it? Is there different behavior in the UDP/HTTP endpoints of Jaeger?

Reply:
There is a bit of asynchronity here. UDP uses `Agent.EmitBatch` with `TCompactProtocol` to send the `ThriftBatch` and HTTP is directly serializing `ThriftBatch` with `TBinaryProtocol` and sending it as byte array ([see Go endpoint](https://github.com/jaegertracing/jaeger/blob/8ed44f2dd9eeb71e323cd4ff93b1fee146fc6442/cmd/collector/app/http_handler.go#L70)). I made the same mistake when doing the `LetsTrace` HttpSender :)